### PR TITLE
Arnej/load direct tensor attribute

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst please review - this is to get the common parts of tensor reading separated out.
@havardpe @toregge FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/tensor/tensor_reader.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <vespa/fastlib/io/bufferedfile.h>
+#include <vespa/searchlib/attribute/readerbase.h>
+#include <vespa/searchlib/util/fileutil.h>
+
+namespace search::tensor {
+
+class TensorReader : public ReaderBase
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/tensor/tensor_reader.h:9]: Please add class comment.

**Review by geirst — APPROVED** [review]
👍

**Comment by arnej27959** [comment]
@geirst @havardpe PTAL

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**